Fix compatibility with newer versions of SvelteKit and error when there is no prettier config file

pmdevita opened the pull request:
### Requirements for Contributing a Bug Fix

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix an existing bug. To contribute other changes, you must use a different template. You can see all templates at <https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE>.
* The pull request must update the test suite to demonstrate the changed functionality. For guidance, please see <https://flight-manual.atom.io/hacking-atom/sections/writing-specs/>.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see <https://github.com/atom/.github/tree/master/CONTRIBUTING.md#pull-requests>.

### Identify the Bug

- SvelteKit changed the API in regards to how prerendered assets are built.
- getPrettierConfig() would attempt to access the `config` property on a potentially null value.

### Description of the Change

- Plugin was updated to use the newer version of the API.
- getPrettierConfig() now checks if the value is null first.

### Verification Process

Packaged it and ran on my own project.

### Release Notes

- Fix compatibility with newer versions of SvelteKit
- Fix error when there is no config file for getPrettierConfig()

pmdevita committed:
Fix adapter compatibility with newer versions of SvelteKit

pmdevita committed:
Fix error returned when plugin is unable to find a prettier config file

idleberg commented:
Oh wow, I was meant to catch up but didn't have time, so this PR is much appreciated! Due to the API change I'd also suggest bumping the SvelteKit version in the manifest. If you don't use `pnpm`, e.g. `npx pnpm install`.

pmdevita commented:
Running `pnpm install` gave me 

```
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

.
└─┬ @sveltejs/eslint-config
  └── ✕ missing peer eslint-plugin-node@">= 11"
Peer dependencies that should be installed:
  eslint-plugin-node@">= 11"  
```

I'm unfamiliar with pnpm but I'm guessing that this dependency is just missing so I added it. `pnpm install eslint-plugin-node`

That fixed the error but `pnpm install` doesn't update SvelteKit (I think this is because of how the version is pinned?). I changed the version to the latest in the package.json (1.0.0-next.391 as of writing), and ran `pnpm install` again, but this then resulted in another peer dependency missing, this time Vite 3.0.0. I added that and now `pnpm install` runs and it should be on the newest version of SvelteKit.

I also fixed the issue where the package `json` was installed instead of `json5`.

pmdevita committed:
Update SvelteKit, fix wrong JSON library, add missing peer dependencies

idleberg commented:
Thanks again for the PR. I will spend some more time looking at the dependency changes and add proper tests before I release a new version.
